fix(parser): fix `SequenceExpression` span

hi-ogawa opened the pull request:
There's a very minor difference in span when `SequenceExpression` has empty spaces right after `(` or before `)`.

For example,  the following code has a different span/range depending on parsers:

```js
( 1, 2 )
  ^^^^    (acorn, swc, babel)
 ^^^^^^   (oxc, tsc)
```

- oxc https://ast.sxzz.dev/#eNpNijEKgDAQBL9ybKUQBC19gC+wvCbECErMBaOiiH/3Cgu7md25EdBitofNbp3SBoOkg5xOySkVVBtqqFQV1ZsjESPLvjrfX8kzWvVFhj0om//dTcFHu3zJKFJtmcHxwfMCI14kXg==
- acorn https://ast.sxzz.dev/#eNotijEOwjAMRa8SeQIpC4w9BQNiymIZD0WpHdlJJVT17hi12/vv/Q0qTPDBFZ1sbh0ytBBIahJMwZd0y+merjE15lYkpQJMC77YfFYpMIWo2Nl7gXx012HEz2/jIy/6HjX4zM3Y2VZ+oLH4/9JtcJEd9h+w8S5G

Just for acorn estree conformance sake, I changed a span to match acorn, but I'm not sure if there's any real value in choosing one over another.

hi-ogawa commented:
* **#9035** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9035?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9035?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9035).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Feb 11, 2:08 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

hi-ogawa committed:
fix(parser): fix `SequenceExpression` span (#9035)

There's a very minor difference in span when `SequenceExpression` has empty spaces right after `(` or before `)`.

For example,  the following code has a different span/range depending on parsers:

```js
( 1, 2 )
  ^^^^    (acorn, swc, babel)
 ^^^^^^   (oxc, tsc)
```

- oxc https://ast.sxzz.dev/#eNpNijEKgDAQBL9ybKUQBC19gC+wvCbECErMBaOiiH/3Cgu7md25EdBitofNbp3SBoOkg5xOySkVVBtqqFQV1ZsjESPLvjrfX8kzWvVFhj0om//dTcFHu3zJKFJtmcHxwfMCI14kXg==
- acorn https://ast.sxzz.dev/#eNotijEOwjAMRa8SeQIpC4w9BQNiymIZD0WpHdlJJVT17hi12/vv/Q0qTPDBFZ1sbh0ytBBIahJMwZd0y+merjE15lYkpQJMC77YfFYpMIWo2Nl7gXx012HEz2/jIy/6HjX4zM3Y2VZ+oLH4/9JtcJEd9h+w8S5G

Just for acorn estree conformance sake, I changed a span to match acorn, but I'm not sure if there's any real value in choosing one over another.
